src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Options/CollectionRuleOptionsExtensions.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Triggers;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.TestCommon.Options
+{
+    internal static class CollectionRuleOptionsExtensions
+    {
+        public static CollectionRuleOptions AddAction(this CollectionRuleOptions options, string type)
+        {
+            return options.AddAction(type, out _);
+        }
+
+        public static CollectionRuleOptions AddAction(this CollectionRuleOptions options, string type, out CollectionRuleActionOptions actionOptions)
+        {
+            actionOptions = new();
+            actionOptions.Type = type;
+
+            options.Actions.Add(actionOptions);
+
+            return options;
+        }
+
+        public static CollectionRuleOptions AddCollectDumpAction(this CollectionRuleOptions options, DumpType type, string egress)
+        {
+            options.AddAction(KnownCollectionRuleActions.CollectDump, out CollectionRuleActionOptions actionOptions);
```

Comment:
Might not be worth calling out, but I noticed it read a bit funny to me so I thought I'd bring it up - we disregard the return value of `AddAction` and just rely on the fact that `options` was manipulated in `AddAction` by being passed by reference (at least that's my best guess). However, in other methods (such as `AddCollectLogsAction` on line 58), we actually do use the returned `options` value. If there's a functional reason for this behavior let me know; I just thought it was interesting that we sometimes chose to use the return value and other times discarded it and used the variable that had been changed by reference.

Reply:
The return type of the `AddAction` and `Add*Action` methods is `CollectionRuleOptions`, which is used to facilitate a builder pattern. The out parameter on these methods (if they have one) is the type of the action options that were added. The return type and the type of the out parameter are distinct.

Reply:
I actually wasn't referring to the out parameter, so maybe I just completely misunderstood what I was looking at. When we call `options.AddAction(...)`, my understanding is that `options` gets updated in the method (for example, `options.Actions.Add(actionOptions)`. Instead of saying `options = options.AddAction(...)`, we don't use the returned value. However, AddCollectLogsAction and AddCollectTraceAction both use the returned `CollectionRuleOptions` in their return statements. Since `options` is passed by reference, functionally, I don't think it makes a difference; I was just wondering why we returned a value if we typically didn't use it.

Reply:
That's the builder pattern.

For example:

```cs
rootOptions.CreateCollectionRule(DefaultRuleName)
    .SetStartupTrigger()
    .AddCollectDumpAction(ExpectedDumpType, ExpectedEgressProvider);
```

can also be written:

```cs
rootOptions.CreateCollectionRule(DefaultRuleName)
    .AddCollectDumpAction(ExpectedDumpType, ExpectedEgressProvider)
    .SetStartupTrigger();
```

I don't want this code to be opinionated on the order in which the rule is constructed, so all of the Add/Set methods return the CollectionRuleOptions to allow you to build it in any order.